test/tests/components/ws-manager/prebuild_test.go

```diff
@@ -414,8 +492,8 @@ func stopWorkspaceAndFindSnapshot(StopWorkspaceFunc integration.StopWorkspaceFun
 	if err != nil {
 		return "", nil, err
 	}
-	if lastStatus == nil && lastStatus.Conditions == nil {
-		return "", nil, nil
+	if lastStatus == nil || lastStatus.Conditions == nil || lastStatus.Conditions.VolumeSnapshot == nil {
+		return "", nil, errors.New("cannot find the last snapshot")
```

Comment:
Is this the root cause for flakey tests?

Reply:
Nope.

Reply:
Oops sorry, I wanted to ask if this code solved flaky.

Reply:
Several reasons for the flaky test happens, but the primary reason is that the prebuild log without the message `🤙 This task ran as a workspace prebuild` in `/workspace/.gitpod/prebuild-log-*`. This message is written by the supervisor when running the prebuild workspace. Somehow, the message does not write into it. I increased the init task with 90 seconds of sleep and hope this will improve the message `🤙 This task ran as a workspace prebuild` be written into the `/workspace/.gitpod/prebuild-log-*`.

If the above issue happens, we fall back to checking the init task message within `/workspace/.gitpod/prebuild-log-*`.

If it still fails, we fall back to checking the someFile generated by the init task exists, this makes sure the workspace pod is created from prebuild, not from the git clone.